envoy.dependency.check/envoy/dependency/check/abstract/release.py

```diff
@@ -22,10 +24,15 @@ def __init__(
             self,
             repo: github.AGithubRepo,
             version: str,
-            release: Optional[github.AGithubRelease] = None) -> None:
+            release: Optional[github.AGithubRelease] = None,
+            session: Optional[aiohttp.ClientSession] = None,
+            url: Optional[str] = None,) -> None:
         self.repo = repo
         self._version = version
         self._release = release
+        self._session = session
+        self._url = url
+        self._sha = None
```

Comment:
im wondering why we want self._sha here - im guessing this was for caching but thats handled by the async_property decorator